src/arrayMethodSort.js

```diff
@@ -4,8 +4,31 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  function compareDefault(a, b) {
+    if (String(a) > String(b)) {
+      return 1;
+    }
+
+    if (String(a) < String(b)) {
+      return -1;
+    }
+
+    return 0;
+  }
+
+  [].__proto__.sort2 = function (compareFunction = compareDefault) {
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = i + 1; j < this.length; j++) {
+        if (compareFunction(this[i], this[j]) > 0) {
+          const temp = this[i];
+
+          this[i] = this[j];
+          this[j] = temp;
+        }
+      }
+    }
+
+    return this;
   };
```

Comment:
The bubble sort algorithm used here is correct, but it is known to be one of the least efficient sorting algorithms for large datasets. If performance is a concern, consider implementing a more efficient sorting algorithm like quicksort or mergesort.